tests/ballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -649,3 +649,9 @@ function test29() returns (boolean, boolean, boolean, boolean, boolean, boolean,
     return (b10, b11, b20, b21, b30, b31, b40, b41);
 }
 
+(function () returns (int)) fb = function () returns (int) { return 100; };
+
+function test30() returns int {
+    int a = fb.call();
+    return a;
+}
```

Comment:
This test doesn't test the scenario you've fixed. The lambda functions are iterated only if the following condition is met:
```
varSymbol.type.tag == TypeTags.FUTURE && ((BFutureType) varSymbol.type).workerDerivative
```
So your test needs to change accordingly.

Reply:
Changed the test accordingly with 07a6575. Thanks :)